### PR TITLE
Consultation form drop events in discord channel via Webhook

### PR DESCRIPTION
**Pull request by Ovifer13**
**Issue #255**

- [x] Use [discord-webhook-node](https://github.com/matthew1232/discord-webhook-node#readme) to create a Webhook that sends a discord message every time a user submits a consultation form.

**Proof**
image

### CODE REVIEW

**Review by sofiaromorales — APPROVED** [review]

**Comment by sofiaromorales** [comment]
Where I can get the `DISCORD_WEBHOOK=` from?

File: interfaces/ConctactInformationProps.ts
Comment (otech47) [inline interfaces/ConctactInformationProps.ts:1]: Fix type in filename

File: next.config.js [hunk 1]
```diff
@@ -11,6 +11,12 @@ const nextConfig = {
         POSTGRES_DB_PORT: process.env.POSTGRES_DB_PORT,
         AWS_ACCESS_KEY_ID: process.env.AWS_ACCESS_KEY_ID,
         AWS_SECRET_ACCESS_KEY: process.env.AWS_SECRET_ACCESS_KEY,
+        DISCORD_WEBHOOK: process.env.DISCORD_WEBHOOK,
+    },
+    webpack5: true,
+    webpack: (config) => {
+        config.resolve.fallback = { fs: false }
```
Comment (otech47) [inline next.config.js:18]: Please add a comment here explaining what this does and why it's needed

File: pages/consultation/index.tsx [hunk 2]
```diff
@@ -54,6 +48,8 @@ const ConsultationPage: NextPage = () => {
     const [contactInformationError, setContactInformationError] = useState(true)
     const [serviceInformationError, setServiceInformationError] = useState(true)
     const [disabledButton, setDisabledButton] = useState(true)
+    const [projectGoals, setProjectGoals] = useState(String)
+    const [constraints, setConstraints] = useState(String)
```
Comment (otech47) [inline pages/consultation/index.tsx:52]: change this to `useState<string>('')`

File: pages/api/webhooks/discord.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    TIMELINE_BUDGET_CONSTRAINTS
+} from '../../../constants/strings'
+
+const { Webhook, MessageBuilder } = require('discord-webhook-node');
+
+const hook = new Webhook(process.env.DISCORD_WEBHOOK);
+
+export default async function sendMessage(
+    conctactInformation: ContactInformationProps, 
+    budget: any,
```
Comment (otech47) [inline pages/api/webhooks/discord.ts:22]: Let's add a type for this `BudgetFormProps`

```
type BudgetFormProps = {
  minBudget: number,
  maxBudget: number
}
```

File: pages/api/webhooks/discord.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    serviceInformation: ServiceInformationFormProps, 
+    projectGoals: String, 
+    constraints: String
+) {
+    try {
+        const embed = new MessageBuilder()
+            .setTitle(NEW_CONSULTATION)
+            .addField(NAME, conctactInformation.name)
+            .addField(EMAIL, conctactInformation.email)
+            .addField(PHONE_NUMBER, conctactInformation.phoneNumber)
```
Comment (otech47) [inline pages/api/webhooks/discord.ts:32]: Fix typo

**Review by otech47 — CHANGES_REQUESTED** [review]

**Comment by otech47** [comment]
@Ovifer13 can you address requested changes so we can get this into production?

File: pages/api/webhooks/discord.ts (hunk 4 above)
Comment (Ovifer13) [reply pages/api/webhooks/discord.ts:32]: Sorry Oscar, what's the typo?

File: pages/api/webhooks/discord.ts (hunk 3 above)
Comment (Ovifer13) [reply pages/api/webhooks/discord.ts:22]: Should I create a new file in the system called "types" where we store this ? Or should I do it how we usually do it with `interface` ?

Comment (otech47) [reply pages/api/webhooks/discord.ts:22]: probably just thesame place as `ContactInformationProps`

**Review by otech47 — APPROVED** [review]

**Pull request merged**